src/main/java/jpa/dao/CompanyDao.java

```diff
@@ -0,0 +1,75 @@
+package jpa.dao;
+
+import jpa.entity.Company;
+import lombok.extern.log4j.Log4j;
+
+import javax.persistence.EntityManager;
+
+@Log4j
+public class CompanyDao {
```

Comment:
better use interface CompanyDao with needed methods
and implement it in class CompanyDaoImpl